aten/src/ATen/mps/MPSGeneratorImpl.mm

```diff
@@ -29,12 +29,19 @@ Generator createMPSGenerator(uint64_t seed_val) {
 
 void MPSGeneratorImpl::set_current_seed(uint64_t seed) {
   data_.seed = seed;
+  data_.offset = 0;
```

Comment:
This is a strange thing to do. Why add offset to MPS generator if you're not actually going to use it for anything. 🤔

Reply:
OK, so it looks like `offset` is not used for anything, but engine does use it. But then that begs the question, why store the offset in two places?

Reply:
This is due is half laziness and other half that I did not want to change the state handling too much (mostly the latter one). `seed` gets the similar treatment, and we store it at 2 places. Both seed and offset are used in get_rng_state and set_rng_state.

But this is not a good answer. Let me see if I can clean this up better.

Reply:
I fixed this in latest commit. Offset state is managed inside the Philox Engine now.